test/e2e/collect-logs.sh

```diff
@@ -34,41 +34,124 @@ echo -e "${GREEN}Starting log collection at ${TIMESTAMP}${NC}"
+# This approach bypasses kubectl logs limitations and accesses the full log history
+# including rotated and compressed log files stored in /var/log/pods.
+collect_node_agent_logs() {
+    local cluster_name=$1
+    local node_log_dir=$2
+    local agent_type=$3  # "hub-agent" or "member-agent"
+
+    echo -e "${YELLOW}Collecting ${agent_type} logs from cluster ${cluster_name} nodes${NC}"
+
+    # Get all nodes in the cluster
+    local nodes
+    nodes=$(kubectl get nodes -o jsonpath='{.items[*].metadata.name}' 2>/dev/null || echo "")
+
+    if [ -z "$nodes" ]; then
+        echo -e "${RED}No nodes found in cluster ${cluster_name}${NC}"
         return
     fi
-    
-    # Collect logs for each container
-    for container in $containers; do
-        log_file="${log_file_prefix}-${container}.log"
-        echo "  - Container ${container} -> ${log_file}"
-        
-        # Get current logs
-        kubectl logs "${pod_name}" -n "${NAMESPACE}" -c "${container}" > "${log_file}" 2>&1 || \
-            echo "Failed to get logs for container ${container}" > "${log_file}"
-        
-        # Try to get previous logs if pod was restarted
-        previous_log_file="${log_file_prefix}-${container}-previous.log"
-        if kubectl logs "${pod_name}" -n "${NAMESPACE}" -c "${container}" --previous > "${previous_log_file}" 2>&1; then
-            echo "  - Previous logs for ${container} -> ${previous_log_file}"
-        else
-            rm -f "${previous_log_file}"
-        fi
+
+    # Create node logs directory
+    mkdir -p "${node_log_dir}"
+
+    for node in $nodes; do
+        echo "  - Collecting ${agent_type} logs from node ${node}"
+        local node_specific_dir="${node_log_dir}/${node}"
+        mkdir -p "${node_specific_dir}"
+
+        # Collect specific agent logs from node filesystem
+        collect_agent_logs_from_node "${node}" "${cluster_name}" "${node_specific_dir}" "${agent_type}"
     done
 }
 
+# Function to collect specific agent logs from node filesystem
+# Collects all log files including rotated (*.log.*) and compressed (*.gz) files
+# Args:
+#   node: The node name to collect logs from
+#   cluster_name: The cluster name for logging context
+#   node_log_dir: The directory to save the collected logs
+#   agent_type: The type of agent ("hub-agent" or "member-agent")
+collect_agent_logs_from_node() {
+    local node=$1
+    local cluster_name=$2
+    local node_log_dir=$3
+    local agent_type=$4  # "hub-agent" or "member-agent"
+
+    echo "    -> Collecting ${agent_type} logs from node filesystem"
+    echo "    -> Found log paths: $(docker exec "${node}" find /var/log/pods -path "*/fleet-system_*${agent_type}*")"
+
+    # First check if any agent logs exist on this node (including .log, .log.*, and .gz files)
+    local log_files
+    log_files=$(docker exec "${node}" find /var/log/pods -path "*/fleet-system_*${agent_type}*" -type f \( -name "*.log" -o -name "*.log.*" -o -name "*.gz" \) 2>/dev/null || echo "")
```

Comment:
can fleet-system be an argument or some constant on top? The fleet-system namespace might not change any time soon but this will help future reader realize that this script depends on fleet-system being the namespace of hub agent and member agent